test/unit/proxy.test.ts

```diff
@@ -102,4 +106,107 @@ describe("proxy", () => {
     expect(resp.status).toBe(200)
     expect(await resp.json()).toBe("coder is the best")
   })
+
+  it("should handle bad requests", async () => {
+    e.use(bodyParser.json({ strict: false }))
+    e.post("/wsup", (req, res) => {
+      res.json(req.body)
+    })
+    codeServer = await integration.setup(["--auth=none"], "")
+    const resp = await codeServer.fetch(proxyPath, {
+      method: "post",
+      body: "coder is the best",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+    expect(resp.status).toBe(400)
+    expect(resp.statusText).toMatch("Bad Request")
+  })
+
+  it("should handle invalid routes", async () => {
+    e.post("/wsup", (req, res) => {
+      res.json(req.body)
+    })
+    codeServer = await integration.setup(["--auth=none"], "")
+    const resp = await codeServer.fetch(`${proxyPath}/hello`)
+    expect(resp.status).toBe(404)
+    expect(resp.statusText).toMatch("Not Found")
+  })
+
+  it("should handle errors", async () => {
+    e.use(bodyParser.json({ strict: false }))
+    e.post("/wsup", (req, res) => {
+      throw new Error("BROKEN")
+    })
+    codeServer = await integration.setup(["--auth=none"], "")
+    const resp = await codeServer.fetch(proxyPath, {
+      method: "post",
+      body: JSON.stringify("coder is the best"),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+    expect(resp.status).toBe(500)
+    expect(resp.statusText).toMatch("Internal Server Error")
+  })
+})
+
+// NOTE@jsjoeio
+// Both this test suite and the one above it are very similar
+// The main difference is this one uses http and node-fetch
+// and specifically tests the proxy in isolation vs. using
+// the httpserver abstraction we've built.
+//
+// Leaving this as a separate test suite for now because
+// we may consider refactoring the httpserver abstraction
+// in the future.
+//
+// If you're writing a test specifically for code in
+// src/node/proxy.ts, you should probably add it to
+// this test suite.
```

Comment:
✍️ @code-asher made a point that the next eng. coming in here might be confused as to where to put their test. I added this comment to address that.

Reply:
TIL that to obtain a random port, we just create and close an arbitrary server around a locking process, and return the port it was assigned:

https://github.com/sindresorhus/get-port/blob/260c667f457c4260574f915c15c470d2c8eaa12d/index.js#L23-L33

Reply:
Nice find! I refactored to use that instead.